File: JavaScript/LinkedList/get_node_value.js
/**
 * Write a function, getNodeVal, that takes in the head of a linked list and an index.
 * The Function should return the value of the linked list at the specified index. If
 * there is no node at the given index, then return null.
 */

export const getNodeVal = (head, index) => {
  let currentNode = head;
  let currentIndex = 0;

  while (currentNode !== null) {
    if (currentIndex === index) return currentNode.data;
    currentNode = currentNode.next;
    currentIndex++;
  }

  return null;
};

// Recursive Approach

export const getNodeValRec = (head, index) => {
  if (head === null) return null;
  if (index === 0) return head.data;

  return getNodeValRec(head.next, index - 1);
};

File: python/LinkedList/get_node_value.py
#!/usr/bin/python3
"""
Write a function, getNodeValue, that takes in the head of a linked list and an index.
The Function should return the value of the linked list at the specified index. If
there is no node at the given index, then return null.
"""

from Node.Node import Node


def get_node_value(head: Node, index:int) -> int | None:
    current_node = head
    counter = 0

    while current_node is not None:
        if counter == index: return current_node.data
        current_node = current_node.next
        counter = counter + 1

    return None



# Recursive Approach

def get_node_value_recursively(head: Node, index: int) -> int | None:

    if head is None: return None
    if index == 0: return head.data

    return get_node_value(head.next, index - 1)
